File: Medium/238-product-of-array-except-self/try-8.cpp
/*
 * Problem #238: Product of Array Except Self
 * Difficulty: Medium
 * Submission: Try 8
 * status: Accepted
 * Language: cpp
 * Date: 24/03/2026, 11:23:59
 * Link: https://leetcode.com/problems/product-of-array-except-self/
 */

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n);
        int current;
        for(int i = 0;i < n; i++){
            if(i == 0) current = 1;
            ans[i] = current;
            current *= nums[i];
        }

         for(int i = n - 1;i >= 0; i--){
            if(i == n - 1) current = 1;
            ans[i] *= current;
            current *= nums[i];
        }
        return ans;
    }
};

File: Easy/121-best-time-to-buy-and-sell-stock/try-8.py
/*
 * Problem #121: Best Time to Buy and Sell Stock
 * Difficulty: Easy
 * Submission: Try 8
 * status: Accepted
 * Language: python3
 * Date: 02/12/2024, 19:02:00
 * Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        L=0
        R=1
        M=0
        while(R<len(prices)):
            dif=prices[R]-prices[L]
            if(prices[L]>prices[R]):
                L=R
            else:
                M=max(M,dif)
            R+=1
        return M

        

            

        
